### PR TITLE
Resource sharing reference

### PR DESCRIPTION
**Pull request by robertkotcher**
will be opening an actual PR in steps

### CODE REVIEW

**Comment by alexec** [comment]
Can I ask you create an enhancement proposal before submitting a PR? It'll allow you to share the Customer Problem, detail Use Cases and generally get support for you idea before investing too much time cutting code.

**Comment by robertkotcher** [comment]
> 

Sorry, yes there

**Comment by robertkotcher** [comment]
> Can I ask you create an enhancement proposal before submitting a PR? It'll allow you to share the Customer Problem, detail Use Cases and generally get support for you idea before investing too much time cutting code.

Sorry, I'm not even sure how this ended up here. I think I meant to pull this back into my local repo! Here's the PR that I actually made, with a link to the issue, as a draft. I will keep this up to date as this conversation evolves.

File: workflow/controller/controller.go [hunk 1]
```diff
@@ -189,6 +194,102 @@ func NewWorkflowController(ctx context.Context, restConfig *rest.Config, kubecli
...
+// todo: test this
+func (wfc *WorkflowController) numActiveResources(namespace, template string) (int, error) {
+	p, err := wfc.kubeclientset.CoreV1().Pods(namespace).List(context.Background(), metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("%s=%s", common.LabelKeyTemplateName, template),
+		FieldSelector: fmt.Sprintf("status.phase=Running"),
+	})
+	if err != nil {
+		return -1, err
+	}
+	log.Error("******")
```
Comment (tooptoop4) [inline workflow/controller/controller.go:216]: 🟢

**Pull request closed without merging**